invenio_vocabularies/contrib/names/mappings/os-v2/names/name-v2.0.0.json

```diff
@@ -0,0 +1,141 @@
+{
+  "settings": {
+    "analysis": {
+      "char_filter": {
+        "strip_special_chars": {
+          "type": "pattern_replace",
+          "pattern": "[\\p{Punct}\\p{S}]",
+          "replacement": ""
+        }
+      },
+      "analyzer": {
+        "accent_analyzer": {
+          "tokenizer": "standard",
+          "type": "custom",
+          "char_filter": ["strip_special_chars"],
+          "filter": ["lowercase", "asciifolding"]
+        }
+      },
+      "filter": {
+        "lowercase": {
+          "type": "lowercase",
+          "preserve_original": true
+        },
+        "asciifolding": {
+          "type": "asciifolding",
+          "preserve_original": true
+        }
+      }
+    }
+  },
+  "mappings": {
+    "dynamic": "strict",
+    "properties": {
+      "$schema": {
+        "type": "keyword",
+        "index": "false"
+      },
+      "created": {
+        "type": "date"
+      },
+      "updated": {
+        "type": "date"
+      },
+      "indexed_at": {
+        "type": "date"
+      },
+      "uuid": {
+        "type": "keyword"
+      },
+      "version_id": {
+        "type": "integer"
+      },
+      "id": {
+        "type": "keyword"
+      },
+      "tags": {
+        "type": "keyword"
+      },
+      "name_sort": {
+        "type": "keyword"
+      },
+      "name": {
+        "type": "text",
+        "analyzer": "accent_analyzer",
+        "search_analyzer": "accent_analyzer",
```

Comment:
Since we're only doing "autocomplete"-style search for names, we don't have to use the analyzer on the "base" text field.

Reply:
So the `search_as_you_type` fields are optimized in such a way that fuzziness cannot be applied on the fields generated by them (`._index_prefix` in this case). So, not removing the analyzer from the base text field for fuzziness functionality to work.

In the recent commit, I have updated the mappings to use edge ngrams and then doing search on the generated tokens instead of search as you type so that fuzziness can be applied.
Looks like dropping the extra sub fields in favor of having ngram tokens is taking up around the same storage with similar search speed.